libraries/chain/util/smt_token.cpp

```diff
@@ -0,0 +1,35 @@
+#include <steem/chain/util/smt_token.hpp>
+
+#ifdef STEEM_ENABLE_SMT
+
+namespace steem { namespace chain { namespace util {
+
+const smt_token_object* smt_token_lookup( database& db, uint32_t nai )
+{
+   const auto& idx = db.get_index< smt_token_index >().indices().get< by_symbol >();
+
+   auto range = idx.range(
```

Comment:
The lower 4 bits are the precision which means `lower_bound( asset_symbol_type::from_nai( nai, 0 ) );` will go straight to the correct value. I think this makes a lot more sense than creating a range that you know only has one element and returning it in a for loop.